### PR TITLE
New lemmas for List.v

### PR DESCRIPTION
**Pull request by ocfnash**
* filter_app (moved from MSets/MSetRBT.v)
 * filter_map
 * filter_ext_in
 * ext_in_filter
 * filter_ext_in_iff
 * filter_ext
 * concat_filter_map
 * combine_nil
 * combine_firstn_l
 * combine_firstn_r
 * combine_firstn
 * nodup_fixed_point

**Kind:** feature

- [x] Corresponding documentation was added / updated (including any warning and error messages added / removed / modified).
- [x] Entry added in the changelog (see https://github.com/coq/coq/tree/master/doc/changelog#unreleased-changelog for details).

Please note that I am essentially a first-time contributor, and as such do not have much insight into what changes are desirable / helpful / worth it. My intention is merely to fill in a few gaps I bumped into while using the standard library; I will not be at all offended if the maintainers feel the best thing would just be for me to close this pull request.

### CODE REVIEW

**Comment by ppedrot** [comment]
@maximedenes summoning the Stdlib2 guy.

File: doc/changelog/10-standard-library/10651-new-lemmas-for-lists.rst
Comment (Zimmi48) [inline doc/changelog/10-standard-library/10651-new-lemmas-for-lists.rst]: Could you please be a bit more specific: new lemmas on `combine`, `filter`, and `nodup` functions on lists. The lemma `filter_app` was moved to the `List` module.

File: theories/MSets/MSetRBT.v
Comment (Zimmi48) [inline theories/MSets/MSetRBT.v]: You need to introduce a deprecated compatibility notation to ensure that this change does not break people's code without a warning:
```
#[deprecated(since="8.11",note="Lemma filter_app has been moved to module List.")]
Notation filter_app := List.filter_app.
```

**Review by Zimmi48 — COMMENTED** [review]
I'd say such additions filling the gaps are always welcome, and mostly unrelated to the stdlib2 effort. I didn't review the statements and proofs though. I'll leave that to the stdlib maintainers.

**Comment by ocfnash** [comment]
Thank you for these remarks @Zimmi48, I have made changes which attempt to follow your suggestions.

File: doc/changelog/10-standard-library/10651-new-lemmas-for-lists.rst
Comment (Zimmi48) [inline doc/changelog/10-standard-library/10651-new-lemmas-for-lists.rst]:
```suggestion
- New lemmas on :g:`combine`, :g:`filter`, and :g:`nodup` functions on lists. The lemma
```

File: doc/changelog/10-standard-library/10651-new-lemmas-for-lists.rst
Comment (Zimmi48) [inline doc/changelog/10-standard-library/10651-new-lemmas-for-lists.rst]:
```suggestion
  :g:`filter_app` was moved to the :g:`List` module.
```

**Review by Zimmi48 — COMMENTED** [review]
Just suggested some quoting.
Note that if you want to use the GitHub interface to apply these suggestions, you can apply both at once using the batch mode only available from the "Files changed" tab.

**Comment by ocfnash** [comment]
@Zimmi48 Thank you again, both for the suggestions and for the GH tip (of which I was unaware).

**Review by herbelin — APPROVED** [review]
Looks very good. Thanks a lot. Does anyone want to add something? Otherwise, I shall merge at the end of the week.

**Comment by ocfnash** [comment]
> Looks very good. Thanks a lot. Does anyone want to add something? Otherwise, I shall merge at the end of the week.

@herbelin I'm delighted you think the proposed changes are useful. Do you still plan to merge in the near future?

File: theories/Lists/List.v
Comment (anton-trunov) [inline theories/Lists/List.v]: Looks like `(f : A -> bool)` is not needed here, since `filter` is essentially `filter f` in this section.

Comment (ocfnash) [reply theories/Lists/List.v]: :man_facepalming: Thanks @anton-trunov I will follow up with a PR this evening.

Comment (ocfnash) [reply theories/Lists/List.v]: I have just created this https://github.com/coq/coq/pull/10731 to address this point.

The PR also includes a proposal to add two more small lemmas which I hope will not prove controversial. I will happily drop from these from the PR if others prefer not to include them.

**Pull request merged**
